src/main/java/org/kohsuke/stapler/idea/language/JellyFileType.java

```diff
@@ -0,0 +1,44 @@
+package org.kohsuke.stapler.idea.language;
+
+import javax.swing.*;
+
+import com.intellij.ide.highlighter.XmlLikeFileType;
+import org.jetbrains.annotations.NonNls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.kohsuke.stapler.idea.icons.Icons;
+
+/**
+ * Jelly file type definition
+ *
+ * @author Julien Greffe
+ */
+public class JellyFileType extends XmlLikeFileType {
+
+    public static final JellyFileType INSTANCE = new JellyFileType();
+
+    public JellyFileType() {
+        super(JellyLanguage.INSTANCE);
+    }
+
+    @Override
+    public @NonNls @NotNull String getName() {
+        return "Jelly";
+    }
+
+    @Override
+    public @NotNull String getDescription() {
+        return "Jelly language file";
```

Comment:
```suggestion
        return "Jelly";
```

I believe this is usually not that much more descriptive than the name. At least, no mention of "file". 
In the case of JetBrains, it is a localized string, but this plugin doesn't have any resource bundle yet.
Also, https://commons.apache.org/proper/commons-jelly/ never describes it as a language. "Executable XML" is the title.